### ISSUE TITLE
Class docstrings are a bit finicky

### ISSUE THREAD

**Issue by Kodiologist** [hylang/hy#1527]
Besides not being documented, they're not recognized if your `defclass` form has an attribute list. The docstring needs to appear in place of the attribute list.

**Comment by gilch** [comment]
For the record, I was opposed to keeping the old attribute list syntax at all. But it does work according to the indented spec: `(defclass <Name> <Base List> [docstring] [attribute list] [class body])`

For example,
```
(defclass Foo [object]
  "doc"
  [x 1]
  (defn foo [self]
    42))
```
```
class Foo(object):
    __doc__ = 'doc'
    x = 1

    def foo(self):
        return 42
```
As you can see, Hy has no problem using both a docstring and an attribute list. This keeps the statements in the same order they're compiled, so naturally the docstring must come before any assignments in the class body.

**Comment by Kodiologist** [comment]
Oh, okay, I misread the compiler method. In any case, it needs to be documented.

**Comment by vodik** [comment]
Off-topic, but +1 with removing the attribute list. Especially if we reintroduce `def` (or some other form) for declaring an attribute with type annotations - then the attribute list ends up also lacking useful functionality.

**Comment by Kodiologist** [comment]
I forgot to close this after it was addressed by #1750 and #1789.
